web app/src/main/java/de/tinf15b4/kino/api/rest/KinoRestController.java

```diff
@@ -0,0 +1,214 @@
+import java.security.SecureRandom;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.google.common.base.Strings;
+
+import de.tinf15b4.kino.data.cinemas.Cinema;
+import de.tinf15b4.kino.data.cinemas.CinemaService;
+import de.tinf15b4.kino.data.favorites.Favorite;
+import de.tinf15b4.kino.data.favorites.FavoriteService;
+import de.tinf15b4.kino.data.movies.Movie;
+import de.tinf15b4.kino.data.movies.MovieService;
+import de.tinf15b4.kino.data.playlists.Playlist;
+import de.tinf15b4.kino.data.playlists.PlaylistService;
+import de.tinf15b4.kino.data.ratedcinemas.RatedCinemaService;
+import de.tinf15b4.kino.data.ratedmovies.RatedMovie;
+import de.tinf15b4.kino.data.ratedmovies.RatedMovieService;
+import de.tinf15b4.kino.data.users.User;
+import de.tinf15b4.kino.data.users.UserService;
+
+@RestController
+public class KinoRestController {
+
+    public static final String NOT_NULL = "Parameters or request body must not be null";
+    public static final String WRONG_PASSWORD = "Wrong username or password";
+    public static final String USER_LOGGED_IN = "This user is already logged in";
+    public static final String TOKEN_INVALID = "Token invalid or expired";
+    public static final String LOGOUT_SUCCESS = "Logout successful";
+    public static final String INVALID_ID = "Id does not exist";
+
+    private ConcurrentHashMap<Token, String> tokens = new ConcurrentHashMap<>();
+    private ConcurrentHashMap<String, Token> users = new ConcurrentHashMap<>();
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private FavoriteService favoriteService;
+
+    @Autowired
+    private CinemaService cinemaService;
+
+    @Autowired
+    private MovieService movieService;
+
+    @Autowired
+    private RatedCinemaService ratedCinemaService;
+
+    @Autowired
+    private RatedMovieService ratedMovieService;
+
+    @Autowired
+    private PlaylistService playlistService;
+
+    @RequestMapping(value = "rest/authorize", method = RequestMethod.GET)
+    public ResponseEntity<?> authorize(@RequestParam(name = "username") String username,
+            @RequestParam(name = "password") String password) throws UnsupportedEncodingException {
+        if (Strings.isNullOrEmpty(username) || Strings.isNullOrEmpty(password))
+            return ResponseEntity.status(HttpStatus.NOT_ACCEPTABLE).body(NOT_NULL);
+
+        username = URLDecoder.decode(username, "UTF-8");
+        password = URLDecoder.decode(password, "UTF-8");
```

Comment:
Does spring not automatically decode parameters? I fear that this might double-decode the parameters. Have you tested usernames containing something like a literal "%20"?